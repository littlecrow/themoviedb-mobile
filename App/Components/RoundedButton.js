import React, { Component } from 'react';
import { TouchableOpacity, Text, View } from 'react-native';
import styles from './Styles/RoundedButtonStyles';
import PropTypes from 'prop-types';

export default class RoundedButton extends Component {
  static propTypes = {
    onPress: PropTypes.func,
    text: PropTypes.string,
    icon: PropTypes.element,
    children: PropTypes.string,
    styles: PropTypes.object
  }

  getText () {
    const buttonText = this.props.text || this.props.children || '';
    return buttonText;
  }

  getIcon () {
    const buttonIcon = this.props.icon || null;
    if (buttonIcon) {
      return <View style={styles.icon}>{buttonIcon}</View>;
    }
    return null;
  }
  render () {
    return (
      <TouchableOpacity style={[styles.button, this.props.styles]} onPress={this.props.onPress}>
        <View style={styles.container}>
          {this.getIcon()}
          <Text style={styles.buttonText}>{this.getText()}</Text>
        </View>
      </TouchableOpacity>
    );
  }
}
