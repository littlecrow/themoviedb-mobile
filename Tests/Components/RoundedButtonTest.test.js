import 'react-native';
import React from 'react';
import RoundedButton from '../../App/Components/RoundedButton';
// import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

test('RoundedButton component renders correctly', () => {
  const _getComponent = (props) => {
    return (
      renderer.create(
        <RoundedButton text={props.text} />
      ).toJSON()
    );
  };

  let component = _getComponent({ text: 'howdy' });
  expect(component).toMatchSnapshot();
});

test('RoundedButton component with children renders correctly', () => {
  const _getComponent = (props) => {
    return (
      renderer.create(
        <RoundedButton>{props.text}</RoundedButton>
      ).toJSON()
    );
  };

  let component = _getComponent({ text: 'Howdy' });
  expect(component).toMatchSnapshot();
});
