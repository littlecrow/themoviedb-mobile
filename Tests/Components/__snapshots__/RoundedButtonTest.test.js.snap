// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoundedButton component renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#2ecc71",
      "borderRadius": 5,
      "height": 40,
      "marginHorizontal": 25,
      "marginVertical": 10,
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#ecf0f1",
          "fontSize": 14,
          "fontWeight": "bold",
        }
      }
    >
      howdy
    </Text>
  </View>
</View>
`;

exports[`RoundedButton component with children renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#2ecc71",
      "borderRadius": 5,
      "height": 40,
      "marginHorizontal": 25,
      "marginVertical": 10,
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#ecf0f1",
          "fontSize": 14,
          "fontWeight": "bold",
        }
      }
    >
      Howdy
    </Text>
  </View>
</View>
`;
